### PR TITLE
feat: tailwind built-in colors support dark mode

### LINKED ISSUE #269: Dark Mode
As someone who often works late at night or in low-light environments, I find that a Dark Mode would be immensely helpful in reducing eye strain and making the application more comfortable to use for extended periods of time. I'm sure many other users would also appreciate this feature.

### PR DESCRIPTION
**Pull request by Innei**
### <samp>🤖 Generated by Copilot at 1e5de14</samp>

This pull request adds a new feature to generate CSS variables from the TailwindCSS color palette and use them for different color schemes. It introduces two new scripts `generate-css-var.js` and `transform.js` that perform the color conversion and output a CSS file `css-var.css` and a JS file `tw-colors.js` with the color values. It also imports the CSS file into the main CSS file `main.css` and exports the JS file as a module.

### <samp>🤖 Generated by Copilot at 1e5de14</samp>

> _`generate-css-var`_
> _Transforms colors for modes_
> _Autumn leaves changing_

### WHY
tailwind built-in colors support dark mode

### <samp>🤖 Generated by Copilot at 1e5de14</samp>

*  Add a new script `generate-css-var.js` that converts the TailwindCSS color palette into CSS variables and output files ([link](https://github.com/Crossbell-Box/xLog/pull/312/files?diff=unified&w=0#diff-28e9fbc4fe9e54b1bd556f0e45a78654bdee169efcef30e919995ebcc9ec3ad0R1-R54))
  * Use the `transform.js` module to create color objects and CSS strings for normal, dark, and light modes ([link](https://github.com/Crossbell-Box/xLog/pull/312/files?diff=unified&w=0#diff-28e9fbc4fe9e54b1bd556f0e45a78654bdee169efcef30e919995ebcc9ec3ad0R1-R54), [link](https://github.com/Crossbell-Box/xLog/pull/312/files?diff=unified&w=0#diff-c8faaae08ad5e864be62a9f7bd328389374b06055c30df988d43f839884c9053R1-R97))
  * Use the `fs` and `path` modules to write the output files `css-var.css` and `tw-colors.js` ([link](https://github.com/Crossbell-Box/xLog/pull/312/files?diff=unified&w=0#diff-28e9fbc4fe9e54b1bd556f0e45a78654bdee169efcef30e919995ebcc9ec3ad0R1-R54))
* Import the generated CSS file `css-var.css` into the main CSS file `main.css` to enable the use of the color variables ([link](https://github.com/Crossbell-Box/xLog/pull/312/files?diff=unified&w=0#diff-36ebfd7bbcd72a8d3e81092b646f4c4427ad2496c2bc1d428c7b5e0ba29430f9R6))
* Add a new file `tw-colors.js` that exports the color object as a JS variable for other uses ([link](https://github.com/Crossbell-Box/xLog/pull/312/files?diff=unified&w=0#diff-efc65fd63c57eb30dc89348f63b486e2c56c45e6e8f79de08a1089322b9415dcR1-R288))

closes #269

### CODE REVIEW

**Comment by Innei** [comment]
Not completion.

image

image

File: package.json [hunk 1]
```diff
@@ -116,7 +118,8 @@
     "unist-util-visit": "4.1.2",
     "use-onclickoutside": "0.4.1",
     "uuid": "9.0.0",
-    "wagmi": "0.12.9"
+    "wagmi": "0.12.9",
+    "zustand": "*"
```
[the remark this answers is not in the document]
Comment (Innei) [reply package.json:122]: zustand 是 @crossbell/connect-kit 的一个 dep，装都装了，装一个版本就好。

File: package.json [hunk 2]
```diff
@@ -119,7 +119,7 @@
     "use-onclickoutside": "0.4.1",
     "uuid": "9.0.0",
     "wagmi": "0.12.9",
-    "zustand": "*"
```
Comment (Innei) [inline package.json]: 他也是 @crossbell/ 依赖，加个 ^ 吧，万一 @crossbell/ 里面的升级了，会出两个 zustand

**Comment by CaliCastle** [comment]
🔥

**Comment by Innei** [comment]
image

image

**Comment by Innei** [comment]
It's ready!

**Comment by Innei** [comment]
image

**Review by sxzz — APPROVED** [review]
It's awesome generally

**Comment by DIYgod** [comment]
This is really great!

And I made a small adjustment to the automatic colors: https://github.com/Crossbell-Box/xLog/commit/5623b7b0799cef23d32d4ecfdeccf59fb17874de

**Pull request merged**